airdrop/airdrop.gno

```diff
@@ -0,0 +1,215 @@
+package airdrop
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"std"
+	"time"
+
+	"gno.land/p/demo/grc/grc20"
+	i256 "gno.land/p/demo/int256"
+	"gno.land/p/demo/merkle"
+	u256 "gno.land/p/demo/uint256"
+)
+
+var (
+	ErrInvalidProof              = errors.New("INVALID_PROOF")
+	ErrInvalidClaimsCount        = errors.New("INVALID_CLAIMS_COUNT")
+	ErrFirstClaimIsNot128        = errors.New("FIRST_CLAIM_MUST_BE_MULT_128")
+	ErrRefundableTimestampNotSet = errors.New("REFUNDABLE_TIMESTAMP_NOT_SET")
+	ErrTooEarlyToRefund          = errors.New("TOO_EARLY_TO_REFUND")
+)
+
+type Claim struct {
+	ID      uint64
+	Claimee string
+	Amount  *u256.Uint // ? u256
+}
+
+func (c Claim) Bytes() []byte {
+	var buf bytes.Buffer
+	idBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(idBytes, c.ID)
+	buf.Write(idBytes)
+	buf.Write([]byte(c.Claimee))
+	amountStr := c.Amount.ToString()
+	buf.Write([]byte(amountStr))
+	return buf.Bytes()
+}
+
+type Config struct {
+	Root                [32]byte
+	RefundableTimestamp uint64 // TODO: timelock
+	RefundTo            std.Address
+}
+
+type Airdrop struct {
+	token         grc20.IGRC20
+	config        Config
+	claimedBitmap map[uint64]*u256.Uint
```

Comment:
In gno, the largest native type to easily compute bit operation is uint64. Cairo used uint128 because it is the largest native type(except the felt252 which is not 2s power). We can use map[uint64]uint64 here and limit the batch claim to 64 claims at once.

Reply:
fixed https://github.com/gnoswap-labs/governance/pull/2/commits/6a690b2eab931534f5ed9292f19608f4b1c39b2f